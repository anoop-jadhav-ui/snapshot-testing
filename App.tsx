import * as React from 'react';
import './style.css';

export default function App() {
  const [todos, setTodos] = React.useState([]);
  const [todoInput, setTodoInputValue] = React.useState('');

  const addTodoItem = () => {
    setTodos([...todos, todoInput]);
    setTodoInputValue('');
  };

  const clearTodoList = () => {
    setTodos([]);
  };

  return (
    <div>
      <h2>Todo List</h2>
      <ul className="todo-items">
        {todos.map((todo) => (
          <li>{todo}</li>
        ))}
      </ul>
      <textarea
        value={todoInput}
        onChange={(e) => setTodoInputValue(e.target.value)}
      />
      <div className="buttons-wrapper">
        <button type="reset" onClick={clearTodoList}>
          Clear
        </button>
        <button onClick={addTodoItem}>Add</button>
      </div>
    </div>
  );
}
