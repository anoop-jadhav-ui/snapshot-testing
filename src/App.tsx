import React, { useState } from 'react';
import './style.css';

const App = () => {
  const [todos, setTodos] = useState<Array<string>>([]);
  const [todoInput, setTodoInputValue] = useState('');

  const addTodoItem = () => {
    if (todoInput) {
      setTodos([...todos, todoInput]);
      setTodoInputValue('');
    }
  };

  const clearTodoList = () => {
    setTodos([]);
  };

  return (
    <div>
      <h2>Todo List</h2>
      <ul className="todo-items">
        {todos.map((todo) => (
          <li>{todo}</li>
        ))}
      </ul>
      <textarea
        data-testid="input"
        value={todoInput}
        onChange={(e) => setTodoInputValue(e.target.value)}
      />
      <div className="buttons-wrapper">
        <button type="reset" onClick={clearTodoList}>
          Clear
        </button>
        <button onClick={addTodoItem}>Add</button>
      </div>
    </div>
  );
}

export default App;