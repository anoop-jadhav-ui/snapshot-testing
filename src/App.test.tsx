import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('<App/>', () => {
  it('should match the snapshot', () => {
    const container = render(<App />);
    expect(container.container).toMatchSnapshot();
  });
});
