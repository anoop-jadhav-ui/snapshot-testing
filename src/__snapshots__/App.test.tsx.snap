// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should match the snapshot 1`] = `
<div>
  <div>
    <h2>
      Todo List
    </h2>
    <ul
      class="todo-items"
    />
    <textarea
      data-testid="input"
    />
    <div
      class="buttons-wrapper"
    >
      <button
        type="reset"
      >
        Clear
      </button>
      <button>
        Add
      </button>
    </div>
  </div>
</div>
`;
